adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1159,14 +1170,45 @@ protected final void onCommandReceived(
         final String commandRequestId = (command.isOneWay() ? "" : command.getRequestId());
         final String topic = String.format("%s/%s/%s/%s/%s/%s", subscription.getEndpoint(), tenantId, deviceId,
                 subscription.getRequestPart(), commandRequestId, command.getName());
-        endpoint.publish(topic, command.getPayload(), qos, false, false);
-        metrics.incrementCommandDeliveredToDevice(subscription.getTenant());
-        LOG.trace("command published to device [tenant-id: {}, device-id: {}, MQTT client-id: {}]",
-                subscription.getTenant(), subscription.getDeviceId(), endpoint.clientIdentifier());
+        LOG.debug("Publishing command to device [tenant-id: {}, device-id: {}, MQTT client-id: {}, QoS: {}]",
+                subscription.getTenant(), subscription.getDeviceId(), subscription.getClientId(),
+                subscription.getQos());
         final Map<String, String> items = new HashMap<>(3);
-        items.put(Fields.EVENT, "command published to device");
-        items.put(Tags.MESSAGE_BUS_DESTINATION.getKey(), topic);
-        items.put(TracingHelper.TAG_QOS.getKey(), qos.toString());
+        items.put(Fields.EVENT, "Publishing command to device");
+        items.put(TracingHelper.TAG_CLIENT_ID.getKey(), subscription.getClientId());
+        items.put(TracingHelper.TAG_QOS.getKey(), subscription.getQos().toString());
+        commandContext.getCurrentSpan().log(items);
+
+        endpoint.publish(topic, command.getPayload(), subscription.getQos(), false, false, sentHandler -> {
+            if (sentHandler.succeeded()) {
+                if (MqttQoS.AT_LEAST_ONCE.equals(subscription.getQos())) {
+                    cmdHandler.addToWaitingForAcknowledgement(sentHandler.result(), subscription, commandContext);
+                } else {
+                    afterCommandPublished(subscription, commandContext);
+                }
+            } else {
+                LOG.error(
```

Comment:
this is an expected outcome (imagine the device suddenly disappearing due to network failure or power loss). No need to wake up the system admins ;-) simply log at DEBUG level